clients/filesystem-hadoop/build.gradle.kts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+}
+
+dependencies {
+  compileOnly(libs.hadoop2.common)
+  implementation(project(":clients:client-java-runtime", configuration = "shadow"))
```

Comment:
I'm not sure whether to package it by introducing `runtime`, which means that users need to introduce both `client-java-runtime-xxx.jar` and `gravitino-filesystem-hadoop-xxx.jar` in the library. But if we directly introduce the `client-java` dependency here, it may conflict with some other dependencies. FYI @jerryshao @yuqi1129

Reply:
Can you explain a bit, I don't quite get your point?

Reply:
This question is actually related to the shade below. What I mean is that we should introduce the client-java dependency in filesystem-hadoop in the following way:
```
  implementation(project(":clients:client-java-runtime", configuration = "shadow"))
```
or:
```
  implementation(project(":api"))
  implementation(project(":common"))
  implementation(project(":clients:client-java"))
```